Tests/Framework/Storage/StorageTests.cs

```diff
@@ -0,0 +1,64 @@
+﻿using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using NUnit.Framework;
+using Microsoft.Xna.Framework.Storage;
+using Microsoft.Xna.Framework;
+using System.ComponentModel.Design;
+
+namespace MonoGame.Tests.Framework
+{
+    [TestFixture]
+    public class StorageDeviceTests
+    {
+        const string MY_GAME = "MyGame";
+
+        [Test]
+        public void IsConnected_ShouldReturnFalse_OnException()
+        {
+            var device = new StorageDevice(null);
+
+            // We haven't opened a container yet, so should be false
+            Assert.IsFalse(device.IsConnected);
+        }
+
+        [Test]
+        public void OpenContainer_ShouldReturnContainer_OnSuccess()
+        {
+            var device = new StorageDevice(PlayerIndex.One);
+            var container = device.OpenContainer(MY_GAME);
+            Assert.IsNotNull(container);
+            Assert.AreEqual(MY_GAME, container.DisplayName);
+            Assert.AreEqual(device, container.StorageDevice);
+            Assert.AreEqual(true, device.IsConnected);
+            Assert.AreNotEqual(0, device.TotalSpace);
+            Assert.AreNotEqual(0, device.FreeSpace);
+        }
+
+        [Test]
+        public void OpenContainer_ShouldThrowException_OnInvalidDisplayName()
+        {
+            var device = new StorageDevice(PlayerIndex.One);
+            Assert.Throws<ArgumentNullException>(() => device.OpenContainer(null), "Value cannot be null. (Parameter 'displayName')");
+            Assert.Throws<ArgumentException>(() => device.OpenContainer(string.Empty), "The value cannot be an empty string. (Parameter 'displayName')");
```

Comment:
So some platforms have limits on what characters are allowed for save storage names.

Should we just allow the code to fail on that platform?  Or should we set some rules on all platforms?

For example on Windows it would have to be limited by names that work as file names disallow `<>:"/\|?*` in a name.

Also should we come up with some name length limit?

Reply:
Done for Android, iOS, Deskop and Windows. Native left untouched, for now.